docker-traceability-plugin/src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java

```diff
@@ -564,18 +567,69 @@ private synchronized void load() {
             LOGGER.log(Level.SEVERE, "Failed to load the configuration from config path = "+config,e);
         }
     }
-    
+
+    /**
+     * Represents the different response formats (JSON, pretty JSON, XML).
+     */
+    private enum ResponseFormat {
+        JSON("application/json;charset=UTF-8", false),
+        PRETTYJSON("application/json;charset=UTF-8", true),
+        XML("text/xml;charset=UTF-8", false);
+
+        private String contentType;
+        
+        private boolean pretty;
+
+        private static final ResponseFormat DEFAULT = JSON;
+
+        private ResponseFormat(final String contentType, final boolean pretty) {
+            this.contentType = contentType;
+            this.pretty = pretty;
+        }
+
+        public static ResponseFormat fromAlias(final String alias) {
+            if (alias == null) {
+                return DEFAULT;
+            }
+            if (alias.equals("json")) {
+                return JSON;
+            } else if (alias.equals("json-pretty")) {
+                return PRETTYJSON;
+            } else if (alias.equals("xml")) {
+                 // Related to https://issues.jenkins-ci.org/browse/JENKINS-28727
+                throw new IllegalStateException("Unsupported format: " + alias);
+            } else {
+                throw new IllegalStateException("Unsupported format: " + alias);
+            }
+        }
+
+        public String getContentType() {
+            return contentType;
+        }
+
+        public boolean getPretty() {
+            return pretty;
+        }
+    }
+
     /**
-     * Serves the JSON response.
-     * @param item Data to be serialized to JSON
-     * @return HTTP response with application/json MIME type
+     * Serves the response and manages its output format in the response.
+     *
+     * @param item Data to be serialized
+     * @param format Format used in the response
+     *
+     * @return HTTP response with MIME type
      */
-    private static HttpResponse toJSONResponse(final Object item) {
+    private static HttpResponse toFormattedResponse(final Object item, final String format) {
         return new HttpResponse() {
             @Override
             public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
-                ObjectMapper mapper = new ObjectMapper(); 
-                rsp.setContentType("application/json;charset=UTF-8");
+                ObjectMapper mapper = new ObjectMapper();
+                ResponseFormat responseFormat = ResponseFormat.fromAlias(format);
+                rsp.setContentType(responseFormat.getContentType());
```

Comment:
maybe I missed something further up - but format could be XML - yet you are sending json here even though it is taged as XML?